builtin/providers/openstack/resource_openstack_lbaas_listener_v2.go

```diff
@@ -0,0 +1,317 @@
+	networkingClient, err := config.networkingV2Client(d.Get("region").(string))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack networking client: %s", err)
+	}
+
+	adminStateUp := d.Get("admin_state_up").(bool)
+	connLimit := d.Get("connection_limit").(int)
+	var sniContainerRefs []string
+	if raw, ok := d.GetOk("sni_container_refs"); ok {
+		for _, v := range raw.([]interface{}) {
+			sniContainerRefs = append(sniContainerRefs, v.(string))
+		}
+	}
+	createOpts := listeners.CreateOpts{
+		Protocol:               listeners.Protocol(d.Get("protocol").(string)),
+		ProtocolPort:           d.Get("protocol_port").(int),
+		TenantID:               d.Get("tenant_id").(string),
+		LoadbalancerID:         d.Get("loadbalancer_id").(string),
+		Name:                   d.Get("name").(string),
+		DefaultPoolID:          d.Get("default_pool_id").(string),
+		Description:            d.Get("description").(string),
+		ConnLimit:              &connLimit,
+		DefaultTlsContainerRef: d.Get("default_tls_container_ref").(string),
+		SniContainerRefs:       sniContainerRefs,
+		AdminStateUp:           &adminStateUp,
+	}
+
+	log.Printf("[DEBUG] Create Options: %#v", createOpts)
+	listener, err := listeners.Create(networkingClient, createOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack LBaaSV2 listener: %s", err)
+	}
+	log.Printf("[INFO] Listener ID: %s", listener.ID)
+
+	log.Printf("[DEBUG] Waiting for Openstack LBaaSV2 listener (%s) to become available.", listener.ID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"PENDING_CREATE"},
+		Target:     []string{"ACTIVE"},
+		Refresh:    waitForListenerActive(networkingClient, listener.ID),
+		Timeout:    2 * time.Minute,
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return err
+	}
+
+	d.SetId(listener.ID)
+
+	return resourceListenerV2Read(d, meta)
+}
+
+func resourceListenerV2Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	networkingClient, err := config.networkingV2Client(d.Get("region").(string))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack networking client: %s", err)
+	}
+
+	listener, err := listeners.Get(networkingClient, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "LBV2 listener")
+	}
+
+	log.Printf("[DEBUG] Retreived OpenStack LBaaSV2 listener %s: %+v", d.Id(), listener)
+
+	d.Set("id", listener.ID)
+	d.Set("name", listener.Name)
+	d.Set("protocol", listener.Protocol)
+	d.Set("tenant_id", listener.TenantID)
+	d.Set("description", listener.Description)
+	d.Set("protocol_port", listener.ProtocolPort)
+	d.Set("admin_state_up", listener.AdminStateUp)
+	d.Set("default_pool_id", listener.DefaultPoolID)
+	d.Set("connection_limit", listener.ConnLimit)
+	d.Set("sni_container_refs", listener.SniContainerRefs)
+	d.Set("default_tls_container_ref", listener.DefaultTlsContainerRef)
```

Comment:
Can you explain what doesn't work correctly if `Computed: false`?  I ask because in testing I:
1) Create a listener w/ TF.
2) Note the "name" field (just calling out this attr arbitrarily) was properly populated.
3) Manually change the listeners name in OpenStack.
4) Call `terraform plan` with no changes to the .tf script
5) Note the plan output wants to change the listeners name from what I manually set to what is in the .tf script:  `name:   "foo" => "tf_test_listener" (forces new resource)`

I do notice that the terraform.tfstate does not show the name as "foo" after the plan.  I tried adding `Computed: true` to the name attr and re-ran through this process with the same results.